src/deadline/job_attachments/api/manifest.py

```diff
@@ -247,3 +260,182 @@ def process_output(status: FileStatus, path: str, output_diff: ManifestDiff):
             process_output(fast_diff_item[1], fast_diff_item[0], output)
 
     return output
+
+
+def _manifest_upload(
+    manifest_file: str,
+    s3_bucket_name: str,
+    manifest_path: str,
+    s3_metadata: Dict[str, Any],
+    boto_session: boto3.Session,
+    logger: ClickLogger = ClickLogger(False),
+):
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+    API to upload a job attachment manifest to the Content Address Storage. Manifests will be
+    uploaded to s3://{s3_bucket_name}/Manifest/{manifest_path} as per the Deadline CAS folder structure.
+    manifest_file: File Path to the manifest file for upload.
+    s3_bucket_name: S3 bucket name.
+    manifest_path: S3 path for the manifest.
+    s3_metadata: Additional S3 file metadata tagged on upload.
+    boto_session: Boto3 session.
+    logger: Click Logger instance to print to CLI as test or JSON.
+    """
+    # Always upload the manifest file to case root /Manifest with the original file name.
+    manifest_path = manifest_path + "/Manifests/" + Path(manifest_file).name
+
+    # S3 uploader.
+    upload = S3AssetUploader(session=boto_session)
+    with open(manifest_file) as manifest:
+        upload.upload_bytes_to_s3(
+            bytes=BytesIO(manifest.read().encode("utf-8")),
+            bucket=s3_bucket_name,
+            key=manifest_path,
+            progress_handler=logger.echo,
+            extra_args=s3_metadata,
+        )
+
+
+def _manifest_download(
+    download_dir: str,
+    farm_id: str,
+    queue_id: str,
+    job_id: str,
+    boto3_session: boto3.Session,
+    step_id: Optional[str] = None,
+    logger: ClickLogger = ClickLogger(False),
+) -> ManifestDownloadResponse:
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+    API to download the Job Attachment manifest for a Job, and optionally dependencies for Step.
+    download_dir: Download directory.
+    farm_id: The Deadline Farm to download from.
+    queue_id: The Deadline Queue to download from.
+    job_id: Job Id to download.
+    boto_session: Boto3 session.
+    step_id: Optional[str]: Optional, download manifest for a step
+    logger: Click Logger instance to print to CLI as test or JSON.
+    return ManifestDownloadResponse Downloaded Manifest data. Contains source S3 key and local download path.
+    """
+
+    # Deadline Client and get the Queue to download.
+    deadline = boto3_session.client("deadline", config=get_default_client_config())
+
+    queue: dict = deadline.get_queue(
```

Comment:
This returns `Queue` no?

Reply:
Nope, this is a boto3 client as per 322, so its just returning a `dict`, not strongly typed.